assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/stable_diffusion_inpainting_mlflow_wrapper.py

```diff
@@ -0,0 +1,127 @@
+        super().__init__()
+        self._pipe = None
+        self._task_type = task_type
+        self._batch_output_folder = None
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """
+        Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        self._batch_output_folder = os.getenv(BatchConstants.BATCH_OUTPUT_PATH, default=False)
+
+        if self._task_type == Tasks.TEXT_TO_IMAGE_INPAINTING.value:
+            try:
+                model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+                self._pipe = StableDiffusionInpaintPipeline.from_pretrained(model_dir)
+                self._pipe.to(get_current_device())
+                logger.info("Model loaded successfully")
+            except Exception as e:
+                logger.error(f"Failed to load the the model. {str(e)}")
+                raise
+        else:
+            raise ValueError(f"invalid task type {self._task_type}")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """
+        Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data: Pandas DataFrame with 3 columns named "prompt", "image", "mask_image" having text
+                           input, initial image and mask image.
+        :type input_data: pd.DataFrame
+        :return: Pandas dataframe with corresponding generated images and NSFW flag.
+                 Images in form of base64 string.
+        :rtype: pd.DataFrame
+        """
+        # Decode the base64 image column
+        images = input_data.loc[:, [MLflowSchemaLiterals.INPUT_COLUMN_IMAGE]].apply(axis=1, func=process_image)
+        images = images.loc[:, 0].apply(func=get_pil_image).tolist()
+
+        mask_images = input_data.loc[:, [MLflowSchemaLiterals.INPUT_COLUMN_MASK_IMAGE]].apply(
+            axis=1, func=process_image
+        )
+        mask_images = mask_images.loc[:, 0].apply(func=get_pil_image).tolist()
+
+        text_prompts = input_data.loc[:, MLflowSchemaLiterals.INPUT_COLUMN_PROMPT].tolist()
+
+        assert len(text_prompts) == len(images) == len(mask_images), (
+            f"Invalid input. Number of text prompt, image and mask image are expected to be same. "
+            f"But, found text prompt length {len(text_prompts)}, image length {len(images)} and "
+            f"mask_image length {len(mask_images)}"
+        )
+
+        generated_images = []
+        nsfw_content = []
+        if self._batch_output_folder:
+            # Batch endpoint
+            for image, mask_image, text_prompt in zip(images, mask_images, text_prompts):
+                output = self._pipe(
```

Comment:
Why not pass all the images together into self._pipe() similar to online endpoint and loop later on the outputs.images to save them to batch_output_folder ?

Reply:
In case of batch, all the inputs are passed at once. it will run OOM if we invoke pipe with all the images. For online endpoint it's up to the user to send the optimal batch size.

Reply:
What is the user input for batch endpoints in inpainting case?
For CLIP, we are using csv files containing base64 image column and label column.
We are controlling the batch size by limiting the number of rows in a single csv file and passing a folder of csv files to the batch endpoint. 
Wondering if that would work here ?

Reply:
@PhaniShekhar : 
- For batch endpoints the batch size is the #CSVs given as input, NOT #rows in the csv. Are you handling it differently for CLIP? 
- The primary reason for looping over images one by one is OOM issue as Madhu mentioned earlier.

Reply:
Yes. so effective batch size to the model would be (min_batch_size in batch input) * (# rows per csv).
In our example, we are using csv files with 10 rows each and min_batch_size of 2. We also called this out and added a comment to change these values in case of OOM.
This would give some sort of a control for the user on the batch size to the model right ?

Reply:
let's say if the max batch size for a particular deployment is 10, user has to divide his data into multiple csv's with 10 rows and put 1 as mini batch size. just wondering if its user friendly??

Reply:
updated to pass all the images at one go for batch also. Will have to update the notebooks to call out the same to users.